File: Sorting/InsertionSort.cpp
// C Implementation
#include<stdio.h>


void insertionSort(int * arr, int size) {
    int i, j, temp;

    for (i = 1; i < size; i++) {
        /* Save the last element */
        temp = arr[i];
        /* Shift the elements to the right when necessary */
        for (j = i - 1; j >= 0 && arr[j] > temp; j--) {
            arr[j + 1] = arr[j];
        }
        /* Insert the saved element */
        arr[j + 1] = temp;
    }
}

void main () {
    static int arr[10] = {10, 2, 7, 9, 8, 6, 3, 5, 4, 1};
    static int size = 10;

    printf("Previous array: ");

    for (int i = 0; i < size; i++) {
        printf("%d ", arr[i]);
    }
    printf("\n");

    insertionSort(arr, size);
    printf("Sorted array: ");

    for (int i = 0; i < size; i++) {
        printf("%d ", arr[i]);
    }
    printf("\n");
}

File: Sorting/InsertionSort.py
#Python Implementation Insertion Sort

def InsertionSort(list):

  #We consider the first element to be already sorted

  for i in range (1, len(list)):

      key = list[i]
      j = i-1

      while key<list[j] and j>=0:

          list[j], list[j+1] = list[j+1], list[j]
          
          j -= 1
      

  return list

  
list = [2, 13, 7, 4, 9, 3, 6]

print(InsertionSort(list))